serialise/src/Codec/Serialise/GeneralisedUTF8.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE BangPatterns #-}
+
+module Codec.Serialise.GeneralisedUTF8 where
+
+import Control.Monad.ST
+import Data.Bits
+import Data.Char
+import Data.Word
+import Data.Primitive.ByteArray
+
+encodeGenUTF8 :: String -> ByteArray
```

Comment:
So we'd generalise this to return off len and whether it is inside or outside the proper UTF8 subset. Perhaps off will always be 0 here so we can ignore that bit.